src/provider/handler/getPosts.js

```diff
@@ -1,18 +1,34 @@
 import axios from 'axios';
 import {mapPost} from './mappers/postMapper';
+import _url from 'url';
 
 export function getPosts(params) {
-  const {url, categories} = params;  
+  const {url} = params;  
   return new Promise((resolve, reject) => {
-
-    if (!categories) {
-      return reject({message: 'must enter at least one category id', statusCode: 500});
+    const aUrl = _url.parse(url);
+    
+    //make sure this is a valid wordpress category url
+    if (!aUrl || aUrl.path.indexOf('/category/') == -1 || aUrl.path.split('/').length < 3) {
+      return reject({message: 'malformed wordpress category page url', 
+                    statusCode: 500});
     }
 
-    //call the WP-API posts endpoint
-    axios.get(`${url}/wp-json/wp/v2/posts?categories=${categories}`).then(response => {
+    let categorySlug = aUrl.path.split('/').pop();
+    const baseUrl = `${aUrl.protocol}//${aUrl.host}`;
+
+    //call the WP-API categories endpoint to get the category id from our input slug
+    axios.get(`${baseUrl}/wp-json/wp/v2/categories?slug=${categorySlug}`).then(response => {
+      if (!response.data || response.data.length == 0 || !response.data[0].id) {
```

Comment:
you're going to do this on all your axios calls. 
So I invite you to add a helpers or utils file that includes a function that wraps this, like : 
```
const fetchUrl = (url, options) => axios.get(url, options)
  .then(response => {
     if (!response.data && ...) {
        throw new Error('....'); // will go to .catch block, so you can customize the error message you return
     }
     return response.data 
  });
```

then you just use
```
fetch(url)
  .then(handleData)
  .catch(error => Promise.reject({ message: '...', code: 500 }))
```

try to use as many small blocks as you can so you can easily compose functions and chain promises. the result is more readable